internal/core/string.rs

```diff
@@ -288,6 +288,24 @@ pub fn format(args: core::fmt::Arguments<'_>) -> SharedString {
     output
 }
 
+/// A trait for converting a value to a [`SharedString`].
+///
+/// This trait is automatically implemented for any type which implements the [`Display`] trait as long as the trait is in scope.
+/// As such, `ToSharedString` shouldn’t be implemented directly: [`Display`] should be implemented instead, and you get the `ToSharedString` implementation for free.
+pub trait ToSharedString {
```

Comment:
Would `#[diagnostic::on_unimplemented]` make sense here, asking to add a display trait implementation?